activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,17 @@
+*   Introduce ApplicationRecord, an Active Record layer supertype
+
+    It's pretty common for folks to monkey patch `ActiveRecord::Base` to
+    work around an issue or introduce extra functionality. Instead of
+    shoving even more stuff in `ActiveRecord::Base`, `ApplicationRecord` can
+    hold all those custom work the apps may need.
```

Comment:
I think we should rephrase the motivation. What I've understood to be the issue is that engines will force configuration changes onto an application, and that there's no way to keep them separate because both we're configuring `ActiveRecord::Base`. That, in turn, results in monkey patches.

You are right about the convenience of where to include extra app wide record functionality.

cc @rafaelfranca